airflow/providers/cncf/kubernetes/utils/pod_manager.py

```diff
@@ -267,7 +267,7 @@ def consume_logs(*, since_time: DateTime | None = None, follow: bool = True) ->
                 time.sleep(1)
 
     def await_container_completion(self, pod: V1Pod, container_name: str) -> None:
-        while not self.container_is_running(pod=pod, container_name=container_name):
+        while self.container_is_running(pod=pod, container_name=container_name):
```

Comment:
same fix was suggest and merged in https://github.com/apache/airflow/pull/23883 but It was reverted in https://github.com/apache/airflow/pull/24479
so just trying the same thing again is probably not going to work?
cc @akakakakakaa  as the original author

Reply:
Hi, thanks, yeah I'm aware. I opened this PR as a Draft (but wasn't aware that still pinged anyone) to see if I could run the CI tests. 

The original PR was reverted due to failing tests, but unfortunately the CI logs and the slack conversations have both fallen victim to the retention. I'm still figuring out the testing flow locally but am still trying to identify any tests that could have failed with the change of this PR.